pkg/template/depgraph.go

```diff
@@ -0,0 +1,117 @@
+package template
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"github.com/pkg/errors"
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+)
+
+type depGraph struct {
+	Dependencies map[string]map[string]struct{}
+}
+
+// these config functions are used to add their dependencies to the depGraph
+func (d *depGraph) funcMap(parent string) template.FuncMap {
+	addDepFunc := func(dep string, _ ...string) string {
+		d.AddDep(parent, dep)
+		return dep
+	}
+
+	return template.FuncMap{
+		"ConfigOption":          addDepFunc,
+		"ConfigOptionIndex":     addDepFunc,
+		"ConfigOptionData":      addDepFunc,
+		"ConfigOptionEquals":    addDepFunc,
+		"ConfigOptionNotEquals": addDepFunc,
```

Comment:
There are registry templates here too now.

Reply:
yeah, but they don't add dependencies within the config page